app/src/main/java/ch/epfl/sdp/peakar/utils/SettingsUtilities.java

```diff
@@ -0,0 +1,77 @@
+package ch.epfl.sdp.peakar.utils;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.content.res.Configuration;
+import android.content.res.Resources;
+import android.util.DisplayMetrics;
+
+import java.util.Locale;
+
+import ch.epfl.sdp.peakar.R;
+
+/**
+ *  Utility class for the Settings Activity, contains utility methods for the settings like the language
+ */
+public final class SettingsUtilities {
+
+    private static final String DEFAULT_LANGUAGE = "en";
+
+    /**
+     * Updates the language depending on the preferences
+     *
+     * @param context context of the application
+     * @param sharedPreferences shared preferences
+     */
+    public static void updateLanguage(Context context, SharedPreferences sharedPreferences){
+        String language =  sharedPreferences.getString(context.getResources().getString(R.string.language_key), DEFAULT_LANGUAGE);
+        SettingsUtilities.setLocale(context, SettingsUtilities.getLanguageCode(language));
+    }
+
+    /**
+     * Get language code from language
+     *
+     * @param language language
+     * @return language code (2 letters)
+     */
+    private static String getLanguageCode(String language){
```

Comment:
I am not sure which is best. Just to raise the question. But what is more scaleable to have this switch statement or would it be possible to have some kind of static final HashMap/Dictionary? What do you think?

Reply:
I think that in our case there are not so many translations so just a simple switch is probably enough.